skyve-web/src/main/java/org/skyve/impl/web/WebUtil.java

```diff
@@ -321,6 +325,61 @@ public static void requestPasswordReset(String customer, String email) throws Ex
 			// set reset password token for all users with the same email address across all customers
 			List<PersistentBean> users = q.beanResults();
 			if (! users.isEmpty()) {
+				// if configured, check the country and if it is on the blacklist/whitelist
+				boolean geoIpBlocked = false;
+				String geoIpMessage = null;
+				if (UtilImpl.IP_INFO_TOKEN != null) {
+					HttpServletRequest request = EXT.getHttpServletRequest();
+					String clientIpAddress = SecurityUtil.getSourceIpAddress(request);
+					Util.LOGGER.info("Checking country for ip " + clientIpAddress);
+					Optional<String> countryCode = new GeoIpService().getCountryCodeForIp(clientIpAddress); // TODO
+					if (countryCode.isPresent()) {
+						String country = countryCode.get();
+						Util.LOGGER.info("Password reset request from country " + country);
+						if(UtilImpl.COUNTRY_CODES != null) {
+							List<String> countryList = Arrays.asList(UtilImpl.COUNTRY_CODES.split("\\|"));
+							// is this country on the list?
+							boolean found = countryList.stream()
+									.anyMatch(i -> i.equalsIgnoreCase(country));
+							if (found) {
+								// check if the list type is a blacklist and ignore the reset if this country is on the deny list
+								if (UtilImpl.COUNTRY_LIST_TYPE.equalsIgnoreCase(AppConstants.COUNTRY_LIST_TYPE_BLACKLIST_ENUMERATION_CODE)) {
+									geoIpMessage = "Password reset request failed because country " + country
+											+ " is on the blacklist. Suspected bot submission for user with email " + email;
+									Util.LOGGER.warning(geoIpMessage);
+
+									// Record GEO IP block (to be recorded in security log)
+									geoIpBlocked = true;
+								}
+							} else {
+								// check if the list type is a white list and ignore the reset if this country is not on the allow list
+								if (UtilImpl.COUNTRY_LIST_TYPE.equalsIgnoreCase(AppConstants.COUNTRY_LIST_TYPE_WHITELIST_ENUMERATION_CODE)) {
```

Comment:
Should this be a not equals check?

Reply:
No - the flow defined here is if a users country is not on the list **AND** the list is a whitelist, fail.